spark/src/main/java/org/apache/iceberg/spark/source/SparkAppenderFactory.java

```diff
@@ -159,6 +168,7 @@ private StructType lazyPosDeleteSparkType() {
               .withPartition(partition)
               .equalityFieldIds(equalityFieldIds)
               .withKeyMetadata(file.keyMetadata())
+              .withSortOrder(sortOrder)
```

Comment:
@yyanyy, I think we cannot mark files with the current sort order id by default as there is no guarantee the incoming records are sorted. If we had an API to request a specific distribution and ordering in a query engine (will be possible in Spark 3.2, for example), only then we could annotate the files.

Reply:
Thank you for the review! Yes I wasn't sure when `sortOrder` will be available from engine. I assumed that this information could be available when engine constructs the appender factory like `SparkAppenderFactory` since it seems that it will be created per task level, so that `sortOrder` can be assigned based on individual task in Spark. But since I don't know the details about each engine I'll revert the changes to each appender factory for now.